docs/agents/chat_agents.md

```diff
@@ -0,0 +1,141 @@
+
+> Can we design an autonomous communicative agent capable of steering the conversation toward task completion with minimal human supervision?
+
+In our current implementation, we consider agents with the following key features:
+- **Role**: along with the goal and content specification, this sets the initial state of an agent, guiding the agent to take actions during the sequential interaction.
+- **Memory**: in-context memory and external memory which allows the agent to infer and learn in a more grounded approach.
+- **Tools**: a set of functions that our agents can utilize to interact with the external world; essentially this gives embodiments to our agents.
+- **Communication**: our framework allows flexible and scalable communication between agents. This is fundamental for the critical research question.
+- **Reasoning**: we will equip agents with different planning and reward (critic) learning abilities, allowing them to optimize task completion in a more guided approach.
+
+
+
+<!-- - (WIP) **Reasoning Ability**: since any goal can be formalized as the outcome of maximizing cumulative rewards, we will equip our agents with policy which they could follow to achieve goals. -->
+
+<!-- We will first start with the single agent setting, where the agent can interact with users, process and store messages, and utilize external tools to generate responses and accomplish tasks. -->
+
+## Quick Start
+Let's first play with a `ChatAgent` instance by simply initialize it with a system message and interact with user messages.
+
+### Step 0: Prepartions
+```python
+from camel.messages import BaseMessage as bm
+from camel.agents import ChatAgent
+```
+
+### Step 1: Define the Role
+Create a system message to define agent's default role and behaviors.
+```python
+sys_msg = bm.make_assistant_message(
+    role_name='stone',
+    content='you are a curious stone wondering about the universe.')
+```
+
+### Step 2: Initialize the Agent
+```python
+agent = ChatAgent(
+    system_message=sys_msg,
+    message_window_size=10,    # [Optional] the length for chat memory
+    )
+```
+### Step 3: Interact with the Agent with `.step()`
+```python
+# Define a user message
+usr_msg = bm.make_user_message(
+    role_name='prof. claude shannon',
+    content='what is information in your mind?')
+
+# Sending the message to the agent
+response = agent.step(usr_msg)
+
+# Check the response (just for illustrative purpose)
+print(response.msgs[0].content)
+>>> information is the resolution of uncertainty.
+```
+Woohoo, your first agent is ready to play with you!
+
+
+## Advanced Features
+
+### Tool Usage
+```python
+# Import the necessary functions
+from camel.functions import MATH_FUNCS, SEARCH_FUNCS
+
+# Initialize the agent with list of functions
+agent = ChatAgent(
+    system_message=sys_msg,        
+    function_list=[*MATH_FUNCS, *SEARCH_FUNCS]
+    )
+
+# Check if functions are enabled
+agent.is_function_calling_enabled()
+>>> True
+```
+
+### Memory
```

Comment:
Is this duplicated?

Reply:
Yes, I think renaming the section would be better. The default agent setting is powerful enough for the `get_started` section, so we can consider memory as an "advanced feature".

Reply:
i just removed the `single_agent.md` and `role_playing.md` from the `Get Started` section.